### PR TITLE
host: Pass config to containerinit via JSON

### PR DESCRIPTION
**Pull request by titanous**
Instead of passing the configuration via arguments, we write it to a JSON file. We were already using JSON for environment variables, and putting the arguments in a struct, so this patch mostly just renames fields.

### CODE REVIEW

File: host/containerinit/init.go [hunk 1]
```diff
@@ -349,37 +347,37 @@ func setupNetworking(args *ContainerInitArgs) error {
...
+	if err := setupNetworking(c); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func getEnv(args *ContainerInitArgs, key string) string {
-	for _, kv := range args.env {
+func getEnv(c *Config, key string) string {
```
Comment (archSeer) [inline host/containerinit/init.go]: We could now change the `ENV` field to be a `map[string]string`, which would then get rid of `getEnv` and [`writeContainerEnv` on the backend](https://github.com/flynn/flynn/blob/75d6dcf8883eb1ae049483353b1013a408585a6e/host/libvirt_lxc_backend.go#L146).

File: host/libvirt_lxc_backend.go [hunk 2]
```diff
@@ -446,58 +446,53 @@ func (l *LibvirtLXCBackend) Run(job *host.Job) (err error) {
...
 		if len(job.Config.Cmd) > 0 {
-			args = append(args, job.Config.Cmd...)
+			config.Args = append(config.Args, job.Config.Cmd...)
 		} else {
-			args = append(args, imageConfig.Cmd...)
+			config.Args = append(config.Args, imageConfig.Cmd...)
 		}
 	}
 
+	g.Log(grohl.Data{"at": "write_env"})
```
Comment (archSeer) [inline host/libvirt_lxc_backend.go]: This should now be `write_config`

Comment (titanous) [inline host/libvirt_lxc_backend.go]: Fixed.

File: host/containerinit/init.go (hunk 1 above)
Comment (titanous) [inline host/containerinit/init.go]: We need it to be a `[]string` when we pass it to `os/exec` so merging here is fine.

**Comment by archSeer** [comment]
LGTM

**Comment by heavenlyhash** [comment]
LGTM

**Pull request merged**